File: datamodel/src/main/java/datamodel/clinical/TumorDetails.java
package datamodel.clinical;

import java.util.List;
import java.util.Set;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class TumorDetails {

    @Nullable
    public abstract String primaryTumorLocation();

    @Nullable
    public abstract String primaryTumorSubLocation();

    @Nullable
    public abstract String primaryTumorType();

    @Nullable
    public abstract String primaryTumorSubType();

    @Nullable
    public abstract String primaryTumorExtraDetails();

    @Nullable
    public abstract Set<String> doids();

    @Nullable
    public abstract TumorStage stage();

    @Nullable
    public abstract Boolean hasMeasurableDisease();

    @Nullable
    public abstract Boolean hasBrainLesions();

    @Nullable
    public abstract Boolean hasActiveBrainLesions();

    @Nullable
    public abstract Boolean hasCnsLesions();

    @Nullable
    public abstract Boolean hasActiveCnsLesions();
    
    @Nullable
    public abstract Boolean hasBoneLesions();

    @Nullable
    public abstract Boolean hasLiverLesions();

    @Nullable
    public abstract Boolean hasLungLesions();

    @Nullable
    public abstract Boolean hasLymphNodeLesions();

    @Nullable
    public abstract List<String> otherLesions();

    @Nullable
    public abstract String biopsyLocation();

}

File: algo/src/main/kotlin/com/hartwig/actin/datamodel/clinical/TumorDetails.kt
package com.hartwig.actin.datamodel.clinical

data class TumorDetails(
    val name: String = "",
    val doids: Set<String>? = null,
    val stage: TumorStage? = null,
    val derivedStages: Set<TumorStage>? = null,
    val hasMeasurableDisease: Boolean? = null,
    val hasBrainLesions: Boolean? = null,
    val hasActiveBrainLesions: Boolean? = null,
    val hasCnsLesions: Boolean? = null,
    val hasActiveCnsLesions: Boolean? = null,
    val hasBoneLesions: Boolean? = null,
    val hasLiverLesions: Boolean? = null,
    val hasLungLesions: Boolean? = null,
    val hasLymphNodeLesions: Boolean? = null,
    val otherLesions: List<String>? = null,
    val biopsyLocation: String? = null
)
